Sources/Starknet/Data/StarknetTypedData.swift

```diff
@@ -0,0 +1,326 @@
+//
+//  TypedData.swift
+//
+//
+//  Created by Bartosz Rybarski on 14/02/2023.
+//
+
+import BigInt
+import Foundation
+
+public enum StarknetTypedDataError: Error {
+    case decodingError
+    case dependencyNotDefined(String)
+    case invalidShortString
+    case encodingError
+}
+
+/// Sign message for off-chain usage. Follows standard proposed [here](https://github.com/argentlabs/argent-x/discussions/14)
+///
+/// ```swift
+/// let typedDataString = """
+/// {
+///     "types": {
+///         "StarkNetDomain": [
+///             {"name": "name", "type": "felt"},
+///             {"name": "version", "type": "felt"},
+///             {"name": "chainId", "type": "felt"},
+///         ],
+///         "Person": [
+///             {"name": "name", "type": "felt"},
+///             {"name": "wallet", "type": "felt"},
+///         ],
+///         "Mail": [
+///             {"name": "from", "type": "Person"},
+///             {"name": "to", "type": "Person"},
+///             {"name": "contents", "type": "felt"},
+///         ],
+///     },
+///     "primaryType": "Mail",
+///     "domain": {"name": "StarkNet Mail", "version": "1", "chainId": 1},
+///     "message": {
+///         "from": {
+///             "name": "Cow",
+///             "wallet": "0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826",
+///         },
+///         "to": {
+///             "name": "Bob",
+///             "wallet": "0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB",
+///         },
+///         "contents": "Hello, Bob!",
+///     },
+/// }
+/// """
+///
+/// let typedData = try JSONDecoder.decode(StarknetTypedData.self, from: typedDataString.data(using: .utf8)!)
+///
+/// let messageHash = try typedData.getMessageHash(accountAddress: "0x1234")
+/// ```
+public struct StarknetTypedData: Codable {
```

Comment:
Can we make `StarknetTypedData` conform to `Equatable` and `Hashable`

Reply:
Sure